kahypar/partition/partitioner.h

```diff
@@ -200,13 +207,20 @@ inline void Partitioner::setupContext(const Hypergraph& hypergraph, Context& con
 
 inline void Partitioner::sanitize(Hypergraph& hypergraph, const Context& context) {
   const auto result = _single_node_he_remover.removeSingleNodeHyperedges(hypergraph);
-  if (context.partition.verbose_output && result.num_removed_single_node_hes > 0) {
+  const HypernodeID num_removed_large_hes = _large_he_remover.removeLargeHyperedges(hypergraph, context);
+  if ( context.partition.verbose_output &&
```

Comment:
I'm not entirely sure whether `_large_he_remover` should be called in `Partitioner::sanitize`. Wouldn't it be more consistent to call it in `Partitioner::preprocess`?

Reply:
The large hyperedge removal should run before pin sparsification and community detection. There are currently two preprocess function: one runs the pin sparsification and then calls the preprocess function that runs the community detection step. If we would add large HE removal step into both preprocessing functions, we have to check if that step was already executed, e.g., if pin sparsification is activated, then run large HE removal in the pin sparsification preprocessing function and not in the community detection step. Otherwise, we have to run it in the community detection step.

I also think that the naming is not good at the moment. I would suggest to rename sanitize -> preprocess, preprocess -> runPinSparsification, preprocess -> runCommunityDetection